crates/iceberg-catalog/src/config.rs

```diff
@@ -99,6 +99,26 @@ pub struct DynAppConfig {
     // ------------- Health -------------
     pub health_check_frequency_seconds: u64,
     pub health_check_jitter_millis: u64,
+
+    // ------------- Vault -------------
+    pub vault: Option<VaultConfig>,
+    // ------------- Secrets -------------
+    pub secret_backend: SecretBackend,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum SecretBackend {
+    Vault,
+    Postgres,
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Redact)]
+pub struct VaultConfig {
```

Comment:
We should add those to the Readme (also secret backend selection)

Reply:
updated readme